src/mal_psi.rs

```diff
@@ -197,62 +191,123 @@ pub fn receiver_2(capital_k: Vec<Fq>, m: Fq, b_i_array: Vec<BigInt<4>>, set_y: &
 }
 
 /// Approximation for an ideal permutation.
-/// Uses AES GCM SIV (https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#AES-GCM-SIV)
-/// \Pi (and \Pi^{-1}) in the paper.
+/// Uses AES
+/// $\Pi$ in the paper.
+/// 
+/// # Arguments
+///
+/// * `elt` - A field element
+/// 
+/// # Outputs
+///
+/// * `permuted` - A field element
+fn pi(elt: Fq) -> Fq {
+    let mut block = field_to_block(elt);
+
+    let cipher = Aes128::new(&AES_KEY.into());
+    cipher.encrypt_block(&mut block);
+
+    let permuted = block_to_field(block);
+    permuted
+}
+
+/// Inverse of the function `pi`.
+/// Uses AES
+/// $\Pi^{-1}$ in the paper.
 /// 
 /// # Arguments
 ///
-/// * `input` - A field element
+/// * `elt` - A field element
 /// 
 /// # Outputs
 ///
 /// * `permuted` - A field element
-fn pi(input: Fq) -> Fq {
-    // for the ideal permutation. because we need a simple fixed permutation, we don't need to change the key or nonce?
-    // TODO: these should be global variables
-    let key = Aes128GcmSiv::generate_key(&mut OsRng);
-    let cipher = Aes128GcmSiv::new(&key);
-    let nonce = Nonce::from_slice(b"unique nonce"); // 96-bits; unique per message
-
-    let input_string: String = std::format!("{input}");
-    // In AES, encryption and decryption are done by the same operation
-    // from: https://docs.rs/aes-gcm-siv/latest/aes_gcm_siv/#usage
-    // also: https://stackoverflow.com/questions/23850486/how-do-i-convert-a-string-into-a-vector-of-bytes-in-rust
-    // TODO: why does it give an error when I run `.decrypt`?
-    // TODO: is is_ok() okay (lol) or should I propagate the error with Result<T, E>?
-    // TODO: test whether applying cipher.encrypt gives you the initial value (how is this possible?)
-    let permuted_bytes = cipher.encrypt(nonce, input_string.as_bytes().as_ref());
-    assert!(permuted_bytes.is_ok());
-    let permuted: Fq = <Fq as PrimeField>::from_le_bytes_mod_order(&permuted_bytes.unwrap());
+fn pi_inverse(elt: Fq) -> Fq {
```

Comment:
this looks the same to me as `pi()`

Reply:
Same as before: there might be an issue with the merge request.